src/UnisonLocal/Page/HomePage.elm

```diff
@@ -1,42 +1,208 @@
+                            ( projectName, branches )
+              }
+            , Cmd.none
+            )
+
+        _ ->
+            ( model, Cmd.none )
+
+
+
+-- EFFECTS
+
+
+fetchProjects :
+    (WebData (List ProjectName) -> msg)
+    -> HttpApi.ApiRequest (List ProjectName) msg
+fetchProjects finishedMsg =
+    LocalApi.projects
+        |> HttpApi.toRequest decodeProjectList (RemoteData.fromResult >> finishedMsg)
+
+
+fetchProjectBranches :
+    (WebData ( ProjectName, List BranchSlug ) -> msg)
+    -> ProjectName
+    -> HttpApi.ApiRequest ( ProjectName, List BranchSlug ) msg
+fetchProjectBranches finishedMsg projectName =
+    let
+        decodeWithProjectName =
+            decodeBranchList
+                |> Decode.map (Tuple.pair projectName)
+    in
+    LocalApi.projectBranches projectName
+        |> HttpApi.toRequest decodeWithProjectName (RemoteData.fromResult >> finishedMsg)
+
+
+
+-- DECODE
+
+
+decodeProjectList : Decode.Decoder (List ProjectName)
+decodeProjectList =
+    Decode.list <|
+        Decode.field "projectName" ProjectName.decode
+
+
+decodeBranchList : Decode.Decoder (List BranchSlug)
+decodeBranchList =
+    let
+        branchSlugDecode =
+            Decode.map BranchRef.branchSlugFromString Decode.string
+                |> Decode.andThen (Util.decodeFailInvalid "Invalid BranchName")
+    in
+    Decode.list <|
+        Decode.field "branchName" branchSlugDecode
+
+
+
+-- VIEW
+
+
+viewProjectList : Projects -> List (Html Msg)
+viewProjectList projects =
+    let
+        branchTag projectName branchName =
+            BranchRef.projectBranchRef branchName
+                |> (\branchRef ->
+                        BranchRef.toTag branchRef
+                            |> Tag.withClick
+                                (Route.projectBranchRoot projectName branchRef Perspective.relativeRootPerspective
+                                    |> Route.toUrlString
+                                    |> Click.href
+                                )
+                   )
+                |> Tag.view
+
+        branchList projectName branches =
+            case branches of
+                [] ->
+                    [ text "No branches" ]
+
+                branchNames ->
+                    branchNames
+                        |> List.map (branchTag projectName)
+                        |> List.intersperse (text " ")
+
+        projectItem projectName branches =
+            div []
+                [ h2 [] [ text <| ProjectName.toString projectName ]
+                , p [] (branchList projectName branches)
+                ]
+    in
+    projects
+        |> Dict.toList
+        |> List.map
+            (\( _, ( projectName, branches ) ) ->
+                projectItem projectName branches
+            )
 
 
 view : Model -> AppDocument Msg
-view _ =
+view { projects } =
     let
         appHeader =
             AppHeader.appHeader
 
+        nonProjectCodeParagraph =
+            p []
```

Comment:
this might be worth styling with `var(--u-color_text_subdued)`; the non-project code is mostly meant to go away. I'm tempted to not have a link to it at all.

Reply:
I've removed the link. It's probably best to avoid the confusion.